test-cypress/integration/explore/class.hierarchy.spec.js

```diff
@@ -1,180 +1,179 @@
                 if (data.name === className) {
-                    let fontSize = $element.prop('style');
-                    expect(fontSize['font-size']).be.eq('10px');
+                    cy.wrap($element).as('classInHierarchy');
                 }
             });
-        // Verify that the diagram zooms that class
-        cy.get('.p-0 > .icon-search')
-            .click()
-            .then(() => {
-                cy.get('#search_input_value').type(className).type('{enter}').wait(500);
+    }
+
+    function verifyClassIsNotExpanded($element) {
+        let fontSize = $element.prop('style');
+        return expect(fontSize['font-size']).be.eq('10px');
+    }
+
+    function verifyClassIsExpanded($element) {
+        let fontSize = $element.prop('style');
+        return expect(fontSize['font-size']).not.be.eq('10px');
+    }
+
+    function confirmReload() {
+        cy.get('.modal-footer .confirm-btn').click();
+        cy.get('.modal').should('not.be.visible');
+    }
+
+    it('Test initial state of the diagram has a class count 50', () => {
+        getCurrentSliderValue()
+            .then((currentValue) => {
+                expect(currentValue === INITIAL_CLASS_COUNT);
             });
+    });
 
-        // Find class ":CabernetSauvignon" and verify changed font-size
-        cy.get('#main-group > text.label')
-            .each(($element) => {
-                let data = $element.prop('__data__');
-                if (data.name === className) {
-                    let fontSize = $element.prop('style');
-                    expect(fontSize['font-size']).not.be.eq('10px');
-                }
-            });
-        // Click on ""Focus diagram""
-        cy.get('.icon-zoom-out').click({force: true}).wait(500);
-        // Verify that the diagram zooms out without resetting the class count.
-        // Select a class ":CabernetSauvignon"
-        cy.get('#main-group > text.label')
-            .each(($element) => {
-                let data = $element.prop('__data__');
-                if (data.name === className) {
-                    let fontSize = $element.prop('style');
-                    expect(fontSize['font-size']).be.eq('10px');
-                }
-            });
+    it('Test show/hide prefixes', () => {
+        // Verify that switching on/off Show/hide prefixes is reflected on the diagram -
+        // prefixes are displayed/hidden
+        verifyPrefixes(($element) => expect($element.text()).to.contain(':'));
+
+        // Switch show prefixes to off
+        cy.get('.toolbar-holder .prefix-toggle-btn').click();
+
+        // Verify that prefixes are removed from diagram
+        verifyPrefixes(($element) => expect($element.text()).to.not.contain(':'));
     });
 
-    it('Test reload diagram', function () {
+    it('Test focus on diagram', () => {
         let className = ':CabernetSauvignon';
+        findClassByName(className);
+        cy.get('@classInHierarchy').then(verifyClassIsNotExpanded);
```

Comment:
This confused me, can't `findClassByName` return the element?

Reply:
It's because of the way `findClassByName` is implemented. As I said in a previous comment, I'll try to refactor it and then it'll make a return as you suggested.

Reply:
As I ended up not refactoring the `findClassByName` this will stay as is. The reason is that it's not possible to return anything else from an `each` loop in cypress. It always yelds the subject on which it was called. It's also not possible to make a local variable store in the function because as we know every cypress command is async and the method will always return before the each loop is completed.